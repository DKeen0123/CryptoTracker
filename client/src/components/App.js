import React from 'react';
import Wrapper from '../containers/Wrapper';
import ApolloClient from 'apollo-boost';
import { ApolloProvider } from 'react-apollo'

const client = new ApolloClient({
  uri: 'http://localhost:4000/graphql'
})

const App = () => {
  return (
    <ApolloProvider client={client}>
      <div>
        <Wrapper />
      </div>
    </ApolloProvider>

  );
};

export default App;
