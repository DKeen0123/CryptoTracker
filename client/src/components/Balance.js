import React from 'react';

const Balance = props => {
  return (
    <div>
      <h3>Balance: {props.userBalance}</h3>
    </div>
  );
};

export default Balance;
