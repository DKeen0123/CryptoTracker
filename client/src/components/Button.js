import React from 'react';

const Button = props => {
  return <button onClick={() => props.transaction(props.label)}>{props.label}</button>;
};

export default Button;
