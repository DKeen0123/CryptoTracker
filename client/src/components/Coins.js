import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchBitcoin } from '../actions/action_bitcoin';
import { fetchEthereum } from '../actions/action_ethereum';
import { gql } from 'apollo-boost';
import { graphql, compose } from 'react-apollo';

const getCryptosQuery = gql`
  {
    cryptoCurrencies{
      name
      amount
      id
    }
  }
`

export class Coins extends Component {
  componentDidMount() {
    this.props.fetchBitcoin();
    this.props.fetchEthereum();
  }

  calculateBitcoin() {
    const { bitcoin } = this.props;
    if (Object.keys(bitcoin).length === 0) return '';
    return this.props.balance / parseInt(bitcoin[0].price_usd, 10);
  }

  calculateEthereum() {
    const { ethereum } = this.props;
    if (ethereum === null || ethereum === undefined) return '';
    if (Object.keys(ethereum).length === 0) return '';
    return this.props.balance / parseInt(ethereum[0].price_usd, 10);
  }

  renderCryptos = () => {
    var data = this.props.data;
    if (data.loading) {
      return( <div>Loading cryptos...</div>)
    } else {
      return (
      data.cryptoCurrencies.map(crypto => <h3 id={crypto.name} key={crypto.id}>{crypto.name}: {crypto.amount}</h3>)
      )
    }
  }

  render() {
    return (
      <div>
        <h3 className="bitcoin-balance">Bitcoin: {this.calculateBitcoin()}</h3>
        <h3 className="ethereum-balance">
          Ethereum: {this.calculateEthereum()}
        </h3>
        {this.renderCryptos()}
      </div>
    );
  }
}

const gqlWrapper = graphql(getCryptosQuery)
const reduxWrapper = connect(state => state, { fetchBitcoin, fetchEthereum })
export default compose(reduxWrapper, gqlWrapper)(Coins);
