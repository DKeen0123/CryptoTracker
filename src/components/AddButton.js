import React from 'react';

const AddButton = props => {
  return <button>{props.label}</button>;
};

export default AddButton;