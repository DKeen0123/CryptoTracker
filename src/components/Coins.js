import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchBitcoin } from '../actions/action_bitcoin';
import { fetchEthereum } from '../actions/action_ethereum';

export class Coins extends Component {
  componentDidMount() {
    this.props.fetchBitcoin();
    this.props.fetchEthereum();
  }

  calculateBitcoin() {
    const { bitcoin } = this.props;
    if (Object.keys(bitcoin).length === 0) return '';
    return this.props.balance / parseInt(bitcoin[0].price_usd, 10);
  }

  calculateEthereum() {
    const { ethereum } = this.props;
    if (ethereum === null || ethereum === undefined) return '';
    if (Object.keys(ethereum).length === 0) return '';
    return this.props.balance / parseInt(ethereum[0].price_usd, 10);
  }

  render() {
    return (
      <div>
        <h3 className="bitcoin-balance">Bitcoin: {this.calculateBitcoin()}</h3>
        <h3 className="ethereum-balance">
          Ethereum: {this.calculateEthereum()}
        </h3>
      </div>
    );
  }
}

export default connect(state => state, { fetchBitcoin, fetchEthereum })(Coins);
