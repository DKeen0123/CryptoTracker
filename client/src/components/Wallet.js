import React, { Component } from 'react';
import { connect } from 'react-redux';
import CryptoPicker from './CryptoPicker';
import Balance from './Balance';
import Button from './Button';
import ChangeBalance from './ChangeBalance';
import { fetchCryptoNames } from '../actions/action_crypto';

export class Wallet extends Component {
  constructor() {
    super();
    this.state = { balance: undefined, chosenCrypto: 'Bitcoin' };
  }

  componentDidMount() {
    this.props.fetchCryptoNames();
  }

  handleUpdateBalance = event =>
    this.setState({ balance: parseInt(event.target.value, 10) });

  transaction = transactionType =>
    transactionType === 'Deposit'? this.deposit() : this.withdraw();

  selectCrypto = event => {
    this.setState({
      chosenCrypto: event.target.value
    });
  }

  // addCrypto = () => this.props.selectedCrypto(this.state.chosenCrypto);


  render() {
    return (
      <div>
        <Balance userBalance={this.props.balance} />
        <ChangeBalance changeBalance={this.handleUpdateBalance} />
        <Button
          label="Deposit"
          className="btn-deposit"
          // transaction={this.transaction}
        />
        <Button
          label="Withdraw"
          className="btn-withdraw"
          // transaction={this.transaction}
        />
        <CryptoPicker
          cryptos={this.props.cryptos}
          selectCrypto={this.selectCrypto}
          chosenCrypto={this.state.chosenCrypto}
          addCrypto={this.addCrypto}
        />
      </div>
    );
  }
}

export default connect(
  state => {
    return { balance: state.balance, cryptos: state.crypto };
  },
  { fetchCryptoNames }
)(Wallet);
