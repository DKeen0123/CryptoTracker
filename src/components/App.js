import React from 'react';
import Wallet from './Wallet';
import Coins from './Coins';
import CryptoPicker from './CryptoPicker';

const App = () => {
  return (
    <div>
      <h2>CryptoTracker</h2>
      <hr />
      <Wallet />
      <CryptoPicker />
      <Coins />
    </div>
  );
};

export default App;
