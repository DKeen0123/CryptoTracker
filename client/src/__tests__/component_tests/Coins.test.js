import React from 'react';
import { shallow, mount } from 'enzyme';
import { Coins } from '../../components/Coins';

describe('Coins', () => {
  let props = {
    balance: 10,
    bitcoin: {},
    data: {
      loading: false,
      cryptoCurrencies: [
        { name: "XRP", id: 1, amount: 0 },
        { name: "TRON", id: 2, amount: 0.3 }
      ]
    }
  };
  let coins = shallow(<Coins {...props} />);

  it('renders correctly', () => {
    expect(coins).toMatchSnapshot();
  });

  describe('renderCryptos', () => {
    it('maps over each and renders them as elements', () => {
      expect(coins.find('#XRP').exists()).toEqual(true)
      expect(coins.find('#TRON').exists()).toEqual(true)
    })
  })

  describe('when mounted', () => {
    const mockFetchBitcoin = jest.fn();
    const mockFetchEthereum = jest.fn();
    beforeEach(() => {
      props.fetchBitcoin = mockFetchBitcoin;
      props.fetchEthereum = mockFetchEthereum;
      coins = mount(<Coins {...props} />);
    });

    it('dispatches the `fetchBitcoin()` method received from props', () => {
      expect(mockFetchBitcoin).toHaveBeenCalled();
    });

    it('dispatches the `fetchEthereum()` method received from props', () => {
      expect(mockFetchEthereum).toHaveBeenCalled();
    });

    describe('when there are valid bitcoin props', () => {
      beforeEach(() => {
        props = {
          balance: 10,
          bitcoin: [{ price_usd: '1000' }],
          data: {
            loading: false,
            cryptoCurrencies: [
              { name: "XRP", id: 1, amount: 0 },
              { name: "TRON", id: 2, amount: 0.3 }
            ]
          }
  };
        coins = shallow(<Coins {...props} />);
      });

      it('renders the correct bitcoin value to the screen', () => {
        expect(coins.find('.bitcoin-balance').text()).toEqual('Bitcoin: 0.01');
      });
    });

    describe('when there are valid ethereum props', () => {
      beforeEach(() => {
        props = {
          balance: 5,
          bitcoin: [{ price_usd: '1000' }],
          ethereum: [{
            price_usd: '500'
          }],
           data: {
            loading: false,
            cryptoCurrencies: [
              { name: "XRP", id: 1, amount: 0 },
              { name: "TRON", id: 2, amount: 0.3 }
            ]
          }
        };
        coins = shallow(<Coins {...props} />);
      });

      it('renders the correct ethereum value to the screen', () => {
        expect(coins.find('.ethereum-balance').text()).toEqual(
          'Ethereum: 0.01'
        );
      });
    });
  });
});
