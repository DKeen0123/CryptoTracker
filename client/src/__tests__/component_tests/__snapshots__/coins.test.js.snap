// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coins renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Coins
    balance={10}
    bitcoin={Object {}}
    data={
        Object {
            "cryptoCurrencies": Array [
              Object {
                "amount": 0,
                "id": 1,
                "name": "XRP",
              },
              Object {
                "amount": 0.3,
                "id": 2,
                "name": "TRON",
              },
            ],
            "loading": false,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="bitcoin-balance"
>
          Bitcoin: 
          
</h3>,
        <h3
          className="ethereum-balance"
>
          Ethereum: 
          
</h3>,
        Array [
          <h3
            id="XRP"
>
            XRP
            : 
            0
</h3>,
          <h3
            id="TRON"
>
            TRON
            : 
            0.3
</h3>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Bitcoin: ",
            "",
          ],
          "className": "bitcoin-balance",
        },
        "ref": null,
        "rendered": Array [
          "Bitcoin: ",
          "",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Ethereum: ",
            "",
          ],
          "className": "ethereum-balance",
        },
        "ref": null,
        "rendered": Array [
          "Ethereum: ",
          "",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "XRP",
            ": ",
            0,
          ],
          "id": "XRP",
        },
        "ref": null,
        "rendered": Array [
          "XRP",
          ": ",
          0,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "TRON",
            ": ",
            0.3,
          ],
          "id": "TRON",
        },
        "ref": null,
        "rendered": Array [
          "TRON",
          ": ",
          0.3,
        ],
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="bitcoin-balance"
>
            Bitcoin: 
            
</h3>,
          <h3
            className="ethereum-balance"
>
            Ethereum: 
            
</h3>,
          Array [
            <h3
              id="XRP"
>
              XRP
              : 
              0
</h3>,
            <h3
              id="TRON"
>
              TRON
              : 
              0.3
</h3>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Bitcoin: ",
              "",
            ],
            "className": "bitcoin-balance",
          },
          "ref": null,
          "rendered": Array [
            "Bitcoin: ",
            "",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Ethereum: ",
              "",
            ],
            "className": "ethereum-balance",
          },
          "ref": null,
          "rendered": Array [
            "Ethereum: ",
            "",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "XRP",
              ": ",
              0,
            ],
            "id": "XRP",
          },
          "ref": null,
          "rendered": Array [
            "XRP",
            ": ",
            0,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "TRON",
              ": ",
              0.3,
            ],
            "id": "TRON",
          },
          "ref": null,
          "rendered": Array [
            "TRON",
            ": ",
            0.3,
          ],
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
